include/task/state_machine.hpp

```diff
@@ -87,7 +89,27 @@ class StateMachineTask : public Task {
    */
   virtual void spin() override;
 
+  /**
+   * @brief Function to get the current state of the state machine
+   */
+  uint8_t get_state();
```

Comment:
Aqui, só pra ficar bonito, acho que dava pra fazer essa função retornar um estado ao invés de um `uint8_t`.

Tipo assim:

```suggestion
  /**
   * @brief Function to get the current state of the state machine
   */
  states get_state();
```

Ou ainda, existe a convenção de colocar `_t` depois de o nome de um tipo, pra ficar claro que é um tipo. Eu prefiro. Daí teria que mudar lá em cima na definição do Enum (não consigo comentar lá pq já tá na develop), mas ficaria tipo

```suggestion
  /**
   * @brief Function to get the current state of the state machine
   */
  state_t get_state();
```

Acho que faz sentido fazermos isso, porque fica mais bonito e já temos o Enum definido lá em cima. Daí a mesma coisa vale pros métodos `get_next_state()`, `get_action()` e pras variáveis membro da classe